src/Jira/Api.php

```diff
@@ -875,14 +882,23 @@ public function getProjectIssueTypes($project_key)
 	/**
 	 * Returns a list of all resolutions.
 	 *
-	 * @param string $project_key Project key.
-	 *
-	 * @return array|false
+	 * @return array
 	 * @since  2.0.0
 	 */
 	public function getResolutions()
 	{
-		return $this->api(self::REQUEST_GET, '/rest/api/2/resolution', array(), true);
+		if ( !count($this->resolutions) ) {
```

Comment:
I usually check such things as `!isset($this->resolutions)`. This way we indeed can handle the case, when `JIRA instance have 0 resolutions defined` VS case, when `we haven't queried resolutions yet`.